docs/reference/scripting/common-script-uses.asciidoc

```diff
@@ -0,0 +1,423 @@
+[[common-script-uses]]
+=== Common scripting use cases
+You can write a script to do almost anything, and sometimes, that's
+the trouble. It's challenging to know what's possible with scripts,
+so the following examples address common uses cases where scripts are
+really helpful.
+
+[[scripting-field-extraction]]
+==== Field extraction
+The goal of this use case is simple; you have fields in your data with a bunch of
```

Comment:
I tend to avoid the word "use case". I think folks don't think of themselves as having a use case. They've got a thing they want to do, but they don't call it a "use case".

Reply:
True -- I think we can just say, "The goal of field extraction is simple..."